### PR TITLE
Error prone fixes

### PR DESCRIPTION
**Pull request by davido**
Activate error prone on Buck by adding these lines to .buckconfig:

```
[tools]
  javac_jar = third-party/error_prone_ant-2.0.15.jar
  compiler_class_name = com.google.errorprone.ErrorProneJavaCompiler
[java]
  extra_arguments = "-Xep:OptionalEquality:OFF"
```

Put `error_prone_ant-2.0.15.jar` to `third-party` directory, run `buck build buck`
and fix the problem flagged by error prone.

### CODE REVIEW

File: src/com/facebook/buck/jvm/java/AccumulateClassNamesStep.java [hunk 1]
```diff
@@ -152,7 +152,7 @@ public InputStream openStream() throws IOException {
             if (existing != null && !existing.equals(value)) {
               throw new IllegalArgumentException(String.format(
                   "Multiple entries with same key but differing values: %1$s=%s and %1$s=%s",
-                  key, value, existing));
+                  key, value));
```
Comment (k21) [inline src/com/facebook/buck/jvm/java/AccumulateClassNamesStep.java]: This and the next change don't look right, I think the format should be changed to something like `%1$s=%2$s and %1$s=%3$s` (or the positional arguments can be removed and we can just pass `key, value, key, existing`).

File: src/com/facebook/buck/util/Escaper.java [hunk 2]
```diff
@@ -410,7 +410,7 @@ public static int decodeNumericEscape(
       boolean valid = false;
       for (int i = 0; i < maxCodeLength && pos < escaped.length(); i++) {
         final char digit = (char) table.indexOf(Character.toLowerCase(escaped.charAt(pos)));
-        if (digit == -1 || digit >= base) {
+        if (digit >= base) {
```
Comment (k21) [inline src/com/facebook/buck/util/Escaper.java]: This looks like it will work correctly, but I would prefer to keep `digit` as an `int` instead.

**Comment by Coneko** [comment]
Oh this is very nice, thanks!

File: src/com/facebook/buck/jvm/java/AccumulateClassNamesStep.java (hunk 1 above)
Comment (davido) [inline src/com/facebook/buck/jvm/java/AccumulateClassNamesStep.java]: @k21 Done. PTAL.

**Comment by davido** [comment]
I addressed all comments.

**Pull request closed without merging**